flowexperimental/comp/wells/CompWellInterface.hpp

```diff
@@ -0,0 +1,108 @@
+/*
+  Copyright 2024, SINTEF Digital
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_COMP_WELLINTERFACE_HPP
+#define OPM_COMP_WELLINTERFACE_HPP
+
+#include <opm/models/utils/propertysystem.hh>
+
+#include <opm/input/eclipse/Schedule/Well/Well.hpp>
+
+#include <string>
+
+#include "SingleCompWellState.hpp"
+
+namespace Opm
+{
+template <typename Scalar>
+class CompConnectionData;
+
+template <typename TypeTag> // TODO: do we need to use TypeTag here?
+class CompWellInterface
```

Comment:
this seems a bit... over the top copy-paste. why do we need a separate interface class for these wells?

Reply:
We do not need at the moment, while naturally we will have both multisegment wells and non-multisegment wells.  Or we might want to test different ways of handling wells.

Reply:
but does the interface need to be different than the generic WellInterface?

Reply:
If you are referring to the `WellInteface.hpp`, that is debatable. The current `WellInterface` has too many things  and we are not sure the way how it works is the optimal way. 

It is good to have a separate play ground in my opinion.  We might be able to refactor in the future to find the common base to be the real `WellInterface`.

Reply:
right. i find it much harder, (and much less likely to happen), when going for distilling proper interfaces than building proper interface bottom-up but different philosophies i guess.

Reply:
I acknowledge your point.

To me the highest priority is to understand how the physics works and get proper results first. And refactoring will be refactoring without changing the correct results (at least knowing the cause if the results need to change).

Reply:
I agree with @akva2 here that it is good to avoid unnecessary code duplication when you can. I suggest you try to use the generic wellInterface and ConnectionData instead of the Comp versions. The rest of the files I would guess are not that trivial to use without changes to the interfaces we use in the blackoil simulator.

Reply:
If only to refer the generic WellInterface, I can have a look. It was a conscious decision to avoid touch the BlackoilWellModel related since that is too big and complicated..